doc/user/content/sql/join.md

```diff
@@ -53,6 +60,45 @@ Instead, `JOIN`s work over the available history of both streams, which
 ultimately provides an experience more similar to an RDBMS than other streaming
 platforms.
 
+### `LATERAL` subqueries
+
+When a subquery appears on the right-hand side of a join, it can optionally be
+preceded by the `LATERAL` keyword. This permits the subquery to refer to columns
+defined by the left-hand side of the join. Normally, a subquery only has access
+to the columns within its own context.
+
+Table functions may also be preceded by the `LATERAL` keyword, although in this
+case the keyword is optional. Table function invocations always have implicit
+access to the columns defined by the left-hand side of the join.
+
+When `LATERAL` behavior is active, the right-hand relation is recomputed for
+each row in the left-hand relation. For example, the following query counts from
+1 to `x` for all the values of `x` in `xs`.
+
+```sql
+SELECT * FROM
+  (VALUES (1), (3)) xs (x)
+  CROSS JOIN LATERAL generate_series(1, x) y;
+```
+```nofmt
+ x | y
+---+---
+ 1 | 1
+ 3 | 1
+ 3 | 2
+ 3 | 3
+```
+
+For a real-world example of a `LATERAL` subquery, see the [Top-K by group
+idiom](/sql/idioms/#top-k-by-group).
+
+{{< warning >}}
```

Comment:
I'd move this warning directly below the line:

> ...each row in the left-hand relation

And then add a paragraph break introducing the example